src/resolve/resolved-link.c

```diff
@@ -299,6 +299,9 @@ static int link_update_dns_servers(Link *l) {
         }
 
         dns_server_unlink_marked(l->dns_servers);
+
+        dns_server_reset_accessible_all(l->dns_servers);
```

Comment:
Maybe also reset in bus_link_method_set_dns_servers_internal() and bus_link_method_revert() ?